### ISSUE TITLE
Scheduling an auto-merge no longer works in 1.18.4

### ISSUE THREAD

**Issue by parnic-sks** [go-gitea/gitea#23022]
### Description

The "auto-merge when checks succeed" feature was working in 1.18.3, but in 1.18.4, users receive

> This pull request is not ready to be merged, check review status and status checks.

Same error message is being seen in #23000 but this is a non-admin scenario, so I thought it probably deserved its own issue.

This is not reproducing on try.gitea.io, but on try, I don't see the auto-merge feature at all, so maybe I'm doing something wrong.

### Gitea Version

1.18.4

### Can you reproduce the bug on the Gitea demo site?

No

### Screenshots

<img width="603" alt="image" src="https://user-images.githubusercontent.com/62021255/220201358-aaaf6bfb-5dd6-4810-9211-4f3733030a92.png">

<img width="602" alt="image" src="https://user-images.githubusercontent.com/62021255/220201413-b05e2b3a-8a08-48e8-b1c8-ca7c9929c644.png">

<img width="867" alt="image" src="https://user-images.githubusercontent.com/62021255/220201450-e350af2f-8726-4c18-9e03-b6e470e79a04.png">

### Git Version

2.39.1

### Operating System

Ubuntu 20.04.3, aarch64/arm64

### How are you running Gitea?

Self-built from 1.18.4 tag with an additional fix to LFS downloads.

### Database

PostgreSQL

**Comment by wxiaoguang** [comment]
The biggest (and somewhat related) PR between 1.18.3 and 1.18.4 is  #22812  @zeripath do you have a time to take a look?

**Comment by wxiaoguang** [comment]
OK, I think I understand the problem now.

It's the same problem as #23000 , the problem is that there were bugs on bug, which made the old code happened to work.

When the "force checking" bug was fixed by https://github.com/go-gitea/gitea/pull/22812, other bugs came.

To fix other bugs, I think the code should be like:

```
skipProtectionCheck := force || auto

CheckPullMergable( ..., skipProtectionCheck bool)
```

A fix is on the way.

**Comment by zeripath** [comment]
Ah bugs that are hidden due to other bugs - the gift that keeps on giving.

I think this clearly indicates we need some tests for the automerge capability - however, that may mean that it needs some restructuring in order to make it testable.

**Comment by lunny** [comment]
I think we need to redesign the automerge, we need a queue like what Github did.

**Comment by lunny** [comment]
Isn't this fixed by #23010 ?

**Comment by zeripath** [comment]
yup Fixed by #23010
